routes/index.js

```diff
@@ -35,8 +35,8 @@ exports.index = function (req, res, next) {
 
 
 exports.admin = function (req, res, next) {
-  console.log(req.body);
-  User.find({ username: req.body.username, password: req.body.password }, function (err, users) {
+  const body = req.body;
+  User.find({ username: body.username, password: body.password }, function (err, users) {
     if (users.length > 0) {
       return res.render('admin', {
         title: 'Admin Access Granted',
```

Comment:
This method is vulnerable to NoSQL injection attacks because it directly queries the MongoDB database with user input without sanitization. Use Mongoose's query validation and consider using parameterized queries.

```suggestion
    User.findOne({ username: username }, function (err, user) {
    if (err) return next(err);

    if (!user) {
      return res.render('admin', {
        title: 'Admin Access',
        granted: false,
      });
    }

    bcrypt.compare(password, user.password, function (err, isMatch) {
      if (err) return next(err);

      if (isMatch) {
        return res.render('admin', {
          title: 'Admin Access Granted',
        });
      } else {
        return res.render('admin', {
          title: 'Admin Access',
          granted: false,
        });
      }
    });
  });
```